index.js

```diff
@@ -29,18 +28,24 @@ module.exports = function (options) {
 }
 
 async function recurse(base, scope, object) {
+  let i = 0;
   scope = _.clone(scope);
   if (_.isArray(object)) return Promise.all(object.map(_.bind(recurse, this, base, scope)));
   else if (_.isPlainObject(object)) {
     if (object["Fn::Map"]) {
       var args = object["Fn::Map"],
         list = args[0],
         placeholder = args[1],
+        idx = `${placeholder}_IDX`,
+        len = `${placeholder}_LEN`,
```

Comment:
LEN should be it's own function, i.e. `Fn::Length` to stay consistent with popular map() functions (https://devdocs.io/javascript/global_objects/array/map)